common/src/java/org/apache/hive/http/HttpServer.java

```diff
@@ -489,13 +496,13 @@ static boolean userHasAdministratorAccess(ServletContext servletContext,
   /**
    * Create the web context for the application of specified name
    */
-  WebAppContext createWebAppContext(Builder b) {
+  WebAppContext createWebAppContext(Builder b) throws FileNotFoundException {
     WebAppContext ctx = new WebAppContext();
     setContextAttributes(ctx.getServletContext(), b.contextAttrs);
     ctx.getServletContext().getSessionCookieConfig().setHttpOnly(true);
     ctx.setDisplayName(b.name);
-    ctx.setContextPath("/");
-    ctx.setWar(appDir + "/" + b.name);
+    ctx.setContextPath(b.contextPath == null ? "/" : b.contextPath);
+    ctx.setWar(getWebAppsPath(b.name) + "/" + b.name);
```

Comment:
do we need this?

Reply:
`ctx.setContextPath(b.contextPath == null ? "/" : b.contextPath);` is not needed, refactored to `ctx.setContextPath(b.contextPath)` because it it always set. This is needed because 2 web apps cannot have the same context path on the same web server. Health check web app has a different context path than WebUI.

`ctx.setWar(getWebAppsPath(b.name) + "/" + b.name);` is needed because each web app has a different app dir. See `service/target/hive-webapps/ `